File: LeetCode_812/Main.py
class Solution(object):
    def largestTriangleArea(self, points):
        """
        :type points: List[List[int]]
        :rtype: float
        """
        def triangleArea(A, B, C):
            return 0.5 * abs(
                A[0] * (B[1] - C[1]) +
                B[0] * (C[1] - A[1]) +
                C[0] * (A[1] - B[1])
            )
        
        n = len(points)
        maxArea = 0.0
        
        for i in range(n):
            for j in range(i + 1, n):
                for k in range(j + 1, n):
                    maxArea = max(maxArea, triangleArea(points[i], points[j], points[k]))
        
        return maxArea

File: LeetCode_812/Main.js
/**
 * @param {number[][]} points
 * @return {number}
 */
const triangleArea = function(x1, y1, x2, y2, x3, y3) {
    // Formula: 1/2 * | x1(y2 - y3) + x2(y3 - y1) + x3(y1 - y2) |
    const area = 0.5 * Math.abs(
        x1 * (y2 - y3) +
        x2 * (y3 - y1) +
        x3 * (y1 - y2)
    );
    return area;
}
var largestTriangleArea = function(points) {
    const n = points.length;
    let area = 0;
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            for (let k = j + 1; k < n; k++) {
                area = Math.max(area, triangleArea(points[i][0], points[i][1], points[j][0], points[j][1], points[k][0], points[k][1],));
            }
        }
    }
    return area;
};